src/QueryM.purs

```diff
@@ -407,6 +413,66 @@ calculateMinFee tx = do
   coinFromEstimate :: FeeEstimate -> Coin
   coinFromEstimate = Coin <<< ((+) (BigInt.fromInt 50000)) <<< unwrap
 
+-- | CborHex-encoded tx
+newtype FinalizedTransaction = FinalizedTransaction ByteArray
+
+derive instance Generic FinalizedTransaction _
+
+instance Show FinalizedTransaction where
+  show = genericShow
+
+instance Json.DecodeJson FinalizedTransaction where
+  decodeJson =
+    map FinalizedTransaction <<<
+      Json.caseJsonString (Left err) (note err <<< hexToByteArray)
+    where
+    err = Json.TypeMismatch "Expected CborHex of Tx"
+
+finalizeTx
+  :: Transaction.Transaction
+  -> Array PlutusData
+  -> Array Transaction.Redeemer
+  -> QueryM (Maybe FinalizedTransaction)
+finalizeTx tx datums redeemers = do
+  -- tx
+  txHex <- liftEffect $
+    byteArrayToHex
+      <<< Serialization.toBytes
+      <<< asOneOf
+      <$> Serialization.convertTransaction tx
+  -- datums
+  encodedDatums <- liftEffect do
+    for datums \datum -> do
+      byteArrayToHex <<< Serialization.toBytes <<< asOneOf
+        <$> maybe' (\_ -> throw $ "Failed to convert plutus data: " <> show datum) pure
+          (Serialization.convertPlutusData datum)
+  -- redeemers
+  encodedRedeemers <- liftEffect $
```

Comment:
I don't think this approach to redeemers is going to work, see my comment above as well